sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -316,6 +321,25 @@ class RocksDB(
     }
   }
 
+  def merge(key: Array[Byte], value: Array[Byte],
+            colFamilyName: String = StateStore.DEFAULT_COL_FAMILY_NAME): Unit = {
+    verifyColFamilyExists(colFamilyName)
+
+    if (conf.trackTotalNumberOfRows) {
+      val oldValue = db.get(colFamilyNameToHandleMap(colFamilyName), readOptions, key)
+      if (oldValue == null) {
+        numKeysOnWritingVersion += 1
+      }
+    }
+    db.merge(colFamilyNameToHandleMap(colFamilyName), writeOptions, key, value)
+
+    if (useColumnFamilies) {
+      changelogWriter.foreach(_.merge(key, value, colFamilyName))
+    } else {
+      changelogWriter.foreach(_.merge(key, value))
```

Comment:
We want to support this for the non-col-families case too ?

Reply:
I agree we probably won't use it in near future, but I don't see a reason to restrict it only to column families. The changes work as expected, and can be used for `default` column family in future if need arises.

Reply:
Yea lets support it for default col family - that could be useful in the future